src/interp.rs

```diff
@@ -0,0 +1,143 @@
+                    self.pc += 1
+                }
+                Instr::PUSH_STR(ref x) => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.push(NativeType::Str(x.clone()));
+                    self.pc += 1
+                }
+                Instr::POP => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.pop();
+                    self.pc +=1
+                }
+                Instr::ADD => panic!("NotYetImplemented"),
+                Instr::SUB => panic!("NotYetImplemented"),
+                Instr::LTEQ => panic!("NotYetImplemented"),
+                Instr::GTEQ => panic!("NotYetImplemented"),
+                Instr::LT => panic!("NotYetImplemented"),
+                Instr::GT => panic!("NotYetImplemented"),
+                Instr::EQEQ => panic!("NotYetImplemented"),
+                Instr::SWAP => panic!("NotYetImplemented"),
+                Instr::DUP => panic!("NotYetImplemented"),
+                Instr::LOAD_VAR(ref index) => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.load_local(index);
+                    self.pc += 1
+                }
+                Instr::STORE_VAR(ref name) => {
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.store_local(name);
+                    self.pc += 1
+                }
+                Instr::LOAD_GLOBAL(ref name) => panic!("NotYetImplemented"),
+                Instr::STORE_GLOBAL(ref name) => panic!("NotYetImplemented"),
+                Instr::NEW_OBJECT(ref class_name) => panic!("NotYetImplemented"),
+                Instr::LOAD_FIELD(ref field_name) => panic!("NotYetImplemented"),
+                Instr::STORE_FIELD(ref field_name) => panic!("NotYetImplemented"),
+                Instr::CALL(ref class_name, ref fn_name) => panic!("NotYetImplemented"),
+                Instr::JUMP_IF_TRUE(ref pos) => panic!("NotYetImplemented"),
+                Instr::JUMP_IF_FALSE(ref pos) => panic!("NotYetImplemented"),
+                Instr::JUMP(ref pos) => panic!("NotYetImplemented"),
+                Instr::RET =>  panic!("NotYetImplemented"),
+                _ => (),
+            };
+        }
+    }
+
+}
+
+struct Frame {
+    stack:  Vec<NativeType>,
+    locals: Vec<NativeType>,
+    return_address: usize
+}
+
+impl Frame {
+    pub fn new(return_address: usize) -> Frame {
+        Frame {
+            stack: Vec::new(),
+            locals: Vec::new(),
+            return_address: return_address,
+        }
+    }
+
+    pub fn push(&mut self, obj: NativeType) {
+        self.stack.push(obj);
+    }
+
+    pub fn pop(&mut self) -> NativeType {
+        match self.stack.pop() {
+            Some(x) => x,
+            None => panic!("Popped from empty stack!"),
+        }
+    }
+
+    fn load_local(&mut self, index: &usize) {
+        let value = self.locals[index.clone()].clone();
```

Comment:
Why `clone` an `&usize`? Indeed, why pass in `&usize` rather than just `usize`?

Reply:
Good point. This fn is called by the match `Instr::LOAD_VAR` arm which passes a `&usize`. It makes sense to change this so that the caller clones the value and passes what the callee needs, without expecting `load_local` to change its signature to accommodate for this

Fixed

Reply:
`usize` is copyable, so doesn't need to be cloned.

Reply:
Ah okay, switch to copy :+1: